From fd2ddc2f8c22251d2170acf9c8d5d4924750456f Mon Sep 17 00:00:00 2001
From: Naveen S <quic_naves@quicinc.com>
Date: Thu, 11 Jul 2024 16:29:39 +0530
Subject: [PATCH] KW: wifi: ath12k: fix NULL pointer access in
 nl80211_del_station function

Currently, we are dereferencing dev pointer without a NULL check
in nl80211_del_station function.
Fix this issue by adding a NULL check.

Patch-dependency: none
Patch-work: none
Fixes: I68c45dee0 (wifi: cfg80211: add support for link id attribute in NL80211_CMD_DEL_STATION)

Signed-off-by: Monika Korada <quic_koramoni@quicinc.com>
Signed-off-by: Naveen S <quic_naves@quicinc.com>
---
 net/wireless/nl80211.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index a013353..a0ac2de 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -8446,10 +8446,12 @@ static int nl80211_del_station(struct sk_buff *skb, struct genl_info *info)
 	int ret;
 	int link_id = nl80211_link_id_or_invalid(info->attrs);
 
+	if (!dev)
+		return -EINVAL;
+
 	memset(&params, 0, sizeof(params));
 
-	if (dev)
-		wdev = dev->ieee80211_ptr;
+	wdev = dev->ieee80211_ptr;
 
 	if (info->attrs[NL80211_ATTR_MAC])
 		params.mac = nla_data(info->attrs[NL80211_ATTR_MAC]);
-- 
2.34.1

